src/modrover/rover.py

```diff
@@ -0,0 +1,48 @@
+from typing import Callable
+
+from .globals import get_rmse
+from .learner import Learner
+from .learnerid import LearnerID
+
+
+class Rover:
+
+    def __init__(
+        self,
+        model_type: str,
+        col_obs: str,
+        col_weights: str,
+        col_fixed: dict[str, list[str]],
+        col_covs: dict[str, list[str]],
+        col_offset: dict[str, str],
+        model_eval_metric: Callable = get_rmse,
+    ) -> None:
+        self.model_type = model_type
+        self.col_obs = col_obs
+        self.col_weights = col_weights
+        self.col_fixed = col_fixed
+        self.col_covs = col_covs
+        self.col_offset = col_offset
+        self.model_eval_metric = model_eval_metric
+
+        # TODO: validuate the inputs
+        # ...
+
+    def get_learner(self, learner_id: LearnerID) -> Learner:
+        all_covariates = list(self.col_covs.values())[0]
```

Comment:
also curious about this magic 0 index. What's special about the first element of col_covs?

Reply:
Here we will assume that `self.col_covs` is a dictionary with only one key corresponding to the original `model_param_name`, base on this assumption, I will just get the covariates from the first and the only value.

Reply:
Change name `col_covs` to `col_covs_explore`